source/common/network/io_socket_handle_impl.cc

```diff
@@ -107,14 +108,51 @@ Api::IoCallUint64Result IoSocketHandleImpl::sendmsg(const Buffer::RawSlice* slic
   message.msg_namelen = address_base->sockAddrLen();
   message.msg_iov = iov.begin();
   message.msg_iovlen = num_slices_to_write;
-  message.msg_control = nullptr;
-  message.msg_controllen = 0;
   message.msg_flags = 0;
-
   auto& os_syscalls = Api::OsSysCallsSingleton::get();
-  const Api::SysCallSizeResult result = os_syscalls.sendmsg(fd_, &message, flags);
-
-  return sysCallResultToIoCallResult(result);
+  if (self_ip == nullptr) {
+    message.msg_control = nullptr;
+    message.msg_controllen = 0;
+    const Api::SysCallSizeResult result = os_syscalls.sendmsg(fd_, &message, flags);
+    return sysCallResultToIoCallResult(result);
+  } else {
+    const int space_v6 = CMSG_SPACE(sizeof(in6_pktinfo));
```

Comment:
nit: prefer unsigned integer variables here and below in similar cases.

Reply:
changed to size_t